### PR TITLE
Update skylab-dev and common packages.yaml (jedi-ufs-env updates)

### LINKED ISSUE #366: Add (NCEPLIBS-)sp to jedi-base-env
**Please describe the package or library you would like to add to spack-stack.**
See https://github.com/JCSDA-internal/soca/issues/817 - `sp` is now a dependency (although optional for some time first) for `gsibec`/`saber`.

**What applications will be using this package or library?**
`saber` when using `gsibec`

**Is there already a package or library in spack-stack that provides this, or related, functionality?**
`sp`

**Additional context**
n/a

**Will This Package be Needed in a NOAA Operational Application?**
Yes - already approved

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

Update `skylab-dev` template and common `packages.yaml` to reflect changes in the associated PR https://github.com/NOAA-EMC/spack/pull/203. 

This change requires some updates in our package configurations, especially the skylab templates, to avoid duplicate packages being built. Same for the Ubuntu CI tests.

## Issues

Fixes https://github.com/NOAA-EMC/spack-stack/issues/366

## Testing

- Testing on Dom's macOS: successfully built skylab-dev environment
- CI testing: Passed for https://github.com/NOAA-EMC/spack-stack/pull/416/commits/60ccc1f9e6d5f26b3f5e55b9e9ce368a09b726a3
- Testing on one HPC with Intel on Discover

### CODE REVIEW

**Comment by climbfuji** [comment]
I think we are slowly getting there with the CI tests passing. But let's merge https://github.com/NOAA-EMC/spack-stack/pull/413 and https://github.com/NOAA-EMC/spack/pull/201 first, assuming that those tests pass, then update this PR and the associated spack PR, update the description in both, and hopefully get a last round of CI tests to pass quickly.

File: .github/actions/setup-os/action.yaml [hunk 1]
```diff
@@ -20,8 +20,8 @@ runs:
         # Install git-lfs to avoid compilation errors of "go" with Intel
         sudo apt-get install git-lfs
 
-        # Install krb5 to avoid compilation errors of "hdf" on Ubuntu
-        sudo apt-get install krb5-user libkrb5-dev
+        ## Install krb5 to avoid compilation errors of "hdf" on Ubuntu
```
Comment (climbfuji) [inline .github/actions/setup-os/action.yaml]: Note to self. Add a comment here that installing krb via apt leads to duplicate packages being built. Do after CI tests ran.

File: configs/templates/skylab-dev/spack.yaml [hunk 2]
```diff
@@ -24,7 +24,7 @@ spack:
 
     # Individual packages
     - bacio@2.4.1
-    - bison@3.8.2
+    #- bison@3.8.2
```
Comment (climbfuji) [inline configs/templates/skylab-dev/spack.yaml:28]: Note to self. Add a comment here that specifying the bison version leads to duplicate packages being built (some using the default/OS bison, some the one requested here). Do after CI tests ran.

File: configs/templates/skylab-dev/spack.yaml [hunk 3]
```diff
@@ -43,7 +43,7 @@ spack:
     - fckit@0.9.5
     # DH* fake version number
     - fiat@1.0.0
-    - flex@2.6.4
+    #- flex@2.6.4
```
Comment (climbfuji) [inline configs/templates/skylab-dev/spack.yaml:48]: Note to self. Add a comment here that specifying the bison version leads to duplicate packages being built (some using the default/OS bison, some the one requested here). Do after CI tests ran.

File: configs/templates/skylab-no-python-dev/spack.yaml [hunk 4]
```diff
@@ -38,7 +38,7 @@ spack:
 
     # Individual packages
     - bacio@2.4.1
-    - bison@3.8.2
+    #- bison@3.8.2
```
Comment (climbfuji) [inline configs/templates/skylab-no-python-dev/spack.yaml:42]: ditto

File: configs/templates/skylab-no-python-dev/spack.yaml [hunk 5]
```diff
@@ -48,19 +48,18 @@ spack:
...
     - eigen@3.4.0
     - esmf@8.3.0b09+pio
     # DH* fake version number
     #- ewok@0.0.1
     - fckit@0.9.5
     # DH* fake version number
     - fiat@1.0.0
-    - flex@2.6.4
-    - fms@2022.01
+    #- flex@2.6.4
```
Comment (climbfuji) [inline configs/templates/skylab-no-python-dev/spack.yaml:63]: ditto

**Review by srherbener — APPROVED** [review]
Tested on my Mac and it works! I can successfully build the stack, then build jedi-bundle. Thanks!

**Pull request merged**